File: gaia-sdk-javascript/src/graphql/response/type/UpdatedBehaviourImpulse.ts

import {Behaviour} from "./Behaviour";

import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../../request/enumeration/Order";
import {OrderByField} from "../../request/enumeration/OrderByField";
import {EdgeOrderByField} from "../../request/enumeration/EdgeOrderByField";

/**
* Impulse which indicates the result of a update behaviour impulse
*/
export interface UpdatedBehaviourImpulse {
    id?:Uuid, 
    /**
    * the behaviour instance
    */
    data?:Behaviour
}

File: gaia-sdk-python/gaia_sdk/graphql/response/type/UpdatedBehaviourImpulse.py

from gaia_sdk.graphql.response.type.Behaviour import Behaviour

Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
Int = int
Boolean = bool


class UpdatedBehaviourImpulse:
    """
    Impulse which indicates the result of a update behaviour impulse
    """
    dictionary: dict

    def __init__(self, dictionary: dict):
        self.dictionary = dictionary

    def __eq__(self, other):
        if type(other) is type(self):
            return self.dictionary == other.dictionary
        return False

    def __repr__(self):
        return {'dictionary': self.dictionary}

    @property
    def id(self) -> Uuid:
        return Uuid(self.dictionary.get("id"))
    """
    the behaviour instance
    """
    @property
    def data(self) -> Behaviour:
        return Behaviour(self.dictionary.get("data"))
